tests/functional/test_sync_wrap.py

```diff
@@ -130,6 +135,32 @@ def _is_rate_limited_polling(response):
         assert False, "Timeout Error: Rate limit with sync wrap request wasn't tripped within set timeout"
 
 
+@pytest.mark.sync_wrap
+@when("the rate limit is tripped with sync-wrap polling the create Patient endpoint")
+def trip_rate_limit_sync_polling_create_patient(context: dict):
+    context["pds"].headers = {
+         "X-Sync-Wait": "29"
+    }
+
+    def _create_patient():
+        response = context["pds"].post(
+            data={"nhsNumberAllocation": "Done"},
+            headers=context["pds"].headers,
+            url=f'{context["pds"].base_url}/Patient/'
+        )
+        return response
+
+    def _is_rate_limited_polling(response):
```

Comment:
Add type hinting.

Reply:
@davidhamill1-nhs Do you want me to add type hinting everywhere else in the file where it's missing, or just for the new code I added?

Reply:
The general rule of thumb we have on the team is to add type hinting to any new class/method/function we create, so don't need to add it to the other places that could/should have type hinting.

Reply:
WRT inner function type hinting, I'll leave it to you to decide.